manager/allocator/network/port/port.go

```diff
@@ -0,0 +1,367 @@
+package port
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/docker/swarmkit/api"
+)
+
+const (
+	// DynamicPortStart is the start of the dynamic port range from which node
+	// ports will be allocated when the user did not specify a port.
+	DynamicPortStart uint32 = 30000
```

Comment:
I think you might be able to get away with an `int16`, although I wouldn't recommend using an unsigned here.

Reply:
Rather, you can just use a regular `int`.

Reply:
I use `uint32` because that's the type of `PortConfig.PublishPort`.